Src/LiquidProjections/EventMap.cs

```diff
@@ -6,63 +6,48 @@ namespace LiquidProjections
 {
     public class EventMap<TContext> : IEventMap<TContext>
     {
-        private readonly Dictionary<Type, List<GetHandlerFor>> mappings = new Dictionary<Type, List<GetHandlerFor>>();
+        private readonly Dictionary<Type, List<Handler>> mappings = new Dictionary<Type, List<Handler>>();
 
         internal CustomHandler<TContext> Do { get; set; }
 
         internal void Add<TEvent>(Func<TEvent, TContext, Task> action)
         {
             if (!mappings.ContainsKey(typeof(TEvent)))
             {
-                mappings[typeof(TEvent)] = new List<GetHandlerFor>();
+                mappings[typeof(TEvent)] = new List<Handler>();
             }
 
-            mappings[typeof(TEvent)].Add(@event => new HandlerWrapper<TEvent>(action).GetHandler(@event));
+            mappings[typeof(TEvent)].Add((@event, context) => action((TEvent)@event, context));
         }
 
         /// <summary>
-        /// Gets an asynchronous handler for <paramref name="event"/> or <c>null</c> if no handler
-        /// has been registered.
+        /// Handles <paramref name="anEvent"/> asynchronously.
         /// </summary>
-        public Func<TContext, Task> GetHandler(object @event)
+        public async Task Handle(object anEvent, TContext context)

```

Comment:
I don't think it makes sense to have another level of indirection here. If someone wants to have a delegate, he can create a new delegate for the `Handle` method itself.